lib/message_mover.rb

```diff
@@ -84,10 +84,29 @@ def bulk_insert_movement_metadata
   # We purposefully omit in_reply_to_id when creating the messages in the
   # new channel, because it could be pointing to a message that has not
   # been moved.
+  #
+  # NOTE: The contortions with ROW_NUMBER along with beginning the created_at
+  # time 5 seconds in the future are to ensure that the moved messages are
+  # always added to the end of the messages stream (which is ordered by
+  # created_at in ChatController) and spaced out appropriately for ordering
+  # in 0.5 second increments, so not all the timestamps appear identical in the UI.
   def create_destination_messages_in_channel(destination_channel)
-    moved_message_ids = DB.query_single(<<~SQL, message_ids: @ordered_source_message_ids, destination_channel_id: destination_channel.id)
+    query_args = {
+      message_ids: @ordered_source_message_ids,
+      destination_channel_id: destination_channel.id,
+      future_start: Time.zone.now + 5.seconds
+    }
+    moved_message_ids = DB.query_single(<<~SQL, query_args)
       INSERT INTO chat_messages(chat_channel_id, user_id, message, cooked, cooked_version, created_at, updated_at)
-      SELECT :destination_channel_id, user_id, message, cooked, cooked_version, CLOCK_TIMESTAMP(), CLOCK_TIMESTAMP()
+      SELECT :destination_channel_id,
+             user_id,
+             message,
+             cooked,
+             cooked_version,
+             :future_start::timestamp + (
+                ROW_NUMBER() OVER (ORDER BY chat_messages.created_at) * interval '1 second' / 2
```

Comment:
`0.5` second intervals is fairly arbitrary, I just thought we don't want it _too_ long because if a lot of messages are moved at once we start getting into the "minutes into the future" territory which is a little confusing.